src/app/payment/PaymentDetail.tsx

```diff
@@ -1,18 +1,34 @@
 'use client'
 import { Button } from '@/components/elements/Button'
 import { useLazyGetTransactionQuery } from '@/redux/api/paymentApi'
+import { transactionErrorStatuses } from '@/types/payment'
 import { useSearchParams } from 'next/navigation'
 import React, { useEffect } from 'react'
+import { PaymentSuccess } from './PaymentSuccess'
+import { PaymentError } from './PaymentError'
+import Link from 'next/link'
 
 export const PaymentDetail = () => {
   const searchParams = useSearchParams()
   const orderId = searchParams.get('order_id')
+  const status = searchParams.get('transaction_status')
+
   const [getTransaction, { data, isLoading }] = useLazyGetTransactionQuery()
   useEffect(() => {
     if (orderId) {
       getTransaction({ order_id: orderId })
     }
   }, [orderId])
+
+  if (status == 'settlement' || status == 'capture')
+    return (
+      <PaymentSuccess
+        status={status ?? ''}
+        packageName={data?.transaction_detail.package.name ?? ''}
+      />
+    )
+  if (transactionErrorStatuses.includes(status ?? ''))
+    return <PaymentError status={status ?? ''} />
```

Comment:
In here, is there really a case where status is null? If there are, I think you could define status as a type of possible values to reduce confusion

Reply:
You're right. Thanks for pointing that outl.